README.md

```diff
@@ -1,94 +1,187 @@
 
-A package of operator code is called a charmed operator or simply "charm".
-You'll use [charmcraft](https://juju.is/docs/sdk/install-charmcraft) to
-register your charm name and publish it when you are ready. You can follow one
-of our [charming tutorials](https://juju.is/docs/sdk/tutorials) to get started
-writing your first charm.
+Note that the `metadata.yaml` file shows that what we have is an example charm called `my-new-charm`, which uses an OCI image resource `httpbin` from `kennethreitz/httpbin`.
 
+Note that the `requirements.txt` file lists a version of `ops`. 
 
-## Testing your charms
+Note that the `src/charm.py` file imports `ops` and uses `ops` constructs to create a charm class `MyNewCharmCharm`, observe Juju events, and pair them to event handlers:
+
+```
+...
+import ops
+...
+class MyNewCharmCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_httpbin_pebble_ready(self, event: ops.PebbleReadyEvent):
+        """Define and start a workload using the Pebble API.
+
+        Change this example to suit your needs. You'll need to specify the right entrypoint and
+        environment configuration for your specific workload.
+
+        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
+        """
+        # Get a reference the container attribute on the PebbleReadyEvent
+        container = event.workload
+        # Add initial Pebble config layer using the Pebble API
+        container.add_layer("httpbin", self._pebble_layer, combine=True)
+        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
+        container.replan()
+        # Learn more about statuses in the SDK docs:
+        # https://juju.is/docs/sdk/constructs#heading--statuses
+        self.unit.status = ops.ActiveStatus()
+...
+```
 
-The framework provides a testing harness, so you can ensure that your charm
-does the right thing in different scenarios, without having to create
-a full deployment. Our [API documentation](https://ops.readthedocs.io/en/latest/#module-ops.testing)
-has the details, including this example:
+> See more: [`ops.PebbleReadyEvent`](https://ops.readthedocs.io/en/latest/index.html#ops.PebbleReadyEvent)
 
-```python
+Note that the `tests/test_charm.py` file imports `ops.testing` and uses it to set up a testing harness:
+
+```
+...
+import ops.testing
+...
 class TestCharm(unittest.TestCase):
-    def test_foo(self):
-        harness = Harness(MyCharm)
-        self.addCleanup(harness.cleanup)  # always clean up after ourselves
+    def setUp(self):
+        self.harness = ops.testing.Harness(MyNewCharmCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+...
```

Comment:
I think we need at least one simple test here, `test_pebble_ready` maybe, to give the flavour of testing with Harness. Let us know if you can't find/add a good one.

Reply:
Will add in more of the content of tests/unit/test_charm.py.